libs/grok-common/src/main/java/org/elasticsearch/grok/Grok.java

```diff
@@ -176,5 +181,39 @@ public boolean match(String text) {
         }
         return null;
     }
+
+    // Code for loading built-in grok patterns packaged with the jar file:
+    private static final String[] PATTERN_NAMES = new String[] {
+        "aws", "bacula", "bro", "exim", "firewalls", "grok-patterns", "haproxy",
+        "java", "junos", "linux-syslog", "mcollective-patterns", "mongodb", "nagios",
+        "postgresql", "rails", "redis", "ruby"
+    };
+
+    public static Map<String, String> loadBuiltinPatterns() throws IOException {
```

Comment:
Why isn't this private? This should only be called once statically right?

Reply:
Right, this should only be called once. Right now this is used from the ingest-common's plugin class. I'll make this private, called once statically and then add another `getBuiltinPatterns()` method that is public, so that ingest-common's Plugin class can use that instead.